glutin/src/api/egl/display.rs

```diff
@@ -124,33 +125,61 @@ impl Display {
             .into());
         }
 
-        let mut attrs = Vec::<EGLint>::with_capacity(2);
+        let mut attrs = Vec::<EGLint>::with_capacity(3);
 
         // TODO: Some extensions exist like EGL_EXT_device_drm which allow specifying
         // which DRM master fd to use under the hood by the implementation. This would
         // mean there would need to be an unsafe equivalent to this function.
 
+        // Push at the end so we can pop it on failure
+        let mut has_display_reference = extensions.contains("EGL_KHR_display_reference");
+        if has_display_reference {
+            attrs.push(egl::TRACK_REFERENCES_KHR as _);
+            attrs.push(egl::TRUE as _);
+        }
+
         // Push `egl::NONE` to terminate the list.
         attrs.push(egl::NONE as EGLint);
 
-        let display = Self::check_display_error(unsafe {
-            egl.GetPlatformDisplayEXT(
-                egl::PLATFORM_DEVICE_EXT,
-                device.raw_device() as *mut _,
-                attrs.as_ptr(),
-            )
-        })
+        // NOTE: This fallback is needed because libglvnd advertises client extensions
+        // if at least one vendor library supports them. This leads to creation
+        // failures for the vendor libraries not supporting
+        // EGL_KHR_display_reference. Also according to the spec creation is allowed
+        // to fail with EGL_KHR_display_reference set to EGL_TRUE even if
+        // EGL_KHR_display_reference is advertised in the client extension
+        // string, so just always try creation without EGL_KHR_display_reference
+        // if it failed using it.
+        let platform_display = loop {
+            let platform_display = Self::check_display_error(unsafe {
+                egl.GetPlatformDisplayEXT(
+                    egl::PLATFORM_DEVICE_EXT,
+                    device.raw_device() as *mut _,
+                    attrs.as_ptr(),
+                )
+            });
+
+            if platform_display.is_err() && has_display_reference {
+                attrs.pop();
+                attrs.pop();
+                attrs.pop();
+                attrs.push(egl::NONE as EGLint);
+                has_display_reference = false;
+                continue;
+            }
+
+            break platform_display;
```

Comment:
Could just do
```suggestion
            match Self::check_display_error(unsafe {
                egl.GetPlatformDisplay(platform, display as *mut _, attrs.as_ptr())
            }) {
                Err(_) if retry_on_failure => {
                    attrs.pop();
                    attrs.pop();
                    attrs.pop();
                    attrs.push(egl::NONE as EGLAttrib);
                    retry_on_failure = false;
                }
                platform_display => break platform_display,
            }
```

This or use `else`, there's no need to do `continue`.